eth2near/eth2near-block-relay-rs/src/hand_made_finality_light_client_update.rs

```diff
@@ -14,19 +14,21 @@ pub struct HandMadeFinalityLightClientUpdate {}
 impl HandMadeFinalityLightClientUpdate {
     pub fn get_finality_light_client_update(
         beacon_rpc_client: &BeaconRPCClient,
-        signature_slot: u64,
+        attested_slot: u64,
     ) -> Result<LightClientUpdate, Box<dyn Error>> {
         const BEACON_STATE_MERKLE_TREE_DEPTH: usize = 5;
         const BEACON_STATE_FINALIZED_CHECKPOINT_INDEX: usize = 20;
 
+        let signature_slot = beacon_rpc_client
+            .get_non_empty_beacon_block_header(attested_slot + 1)?
+            .slot
+            .into();
         let signature_beacon_body =
             beacon_rpc_client.get_beacon_block_body_for_block_id(&format!("{}", signature_slot))?;
         let sync_committe_signature = signature_beacon_body
             .sync_aggregate()
             .map_err(|_| MissSyncAggregationError)?;
 
-        let attested_slot = signature_beacon_body.attestations()[0].data.slot;
```

Comment:
I think we can't just remove this part of code because it is not guaranteed that next non-empty beacon block will sign exactly this attested slot (not for example previous one because this one it doesn't like).

Reply:
The attestations get included one block later, the edge case when the next block is empty, so it will be the first none empty beacon block. 
Also note tha, the first none empty block contains attestation for the last none empty block + for all empty blocks.
Example:
https://lodestar-kiln.chainsafe.io/eth/v1/beacon/blocks/4941

Also please check this:
https://kb.beaconcha.in/attestation